common/core/desktop/src/kmx/kmx_capslock.cpp

```diff
@@ -2,57 +2,88 @@
   Copyright:        Copyright (C) 2003-2018 SIL International.
   Authors:          mcdurdin
 */
+#include <keyman/keyboardprocessor_consts.h>
 #include <kmx/kmx_processevent.h>
 
 using namespace km::kbp;
 using namespace kmx;
 
-void KMX_ProcessEvent::ResetCapsLock(void)
-{
+KMX_BOOL KMX_ProcessEvent::IsCapsLockOn(KMX_DWORD modifiers) {
+  return modifiers & CAPITALFLAG;
+}
+
+void KMX_ProcessEvent::SetCapsLock(KMX_DWORD &modifiers, KMX_BOOL capsLockOn) {
+  m_actions.QueueAction(QIT_CAPSLOCK, capsLockOn);
```

Comment:
done

Reply:
Wondering though - should `capsLockOn` be `bool` instead of `KMX_BOOL` since this is a private method (same for the other  methods below)?

Reply:
Not sure. `KMX_BOOL` is there for the KMX processor because we ported from Windows and wanted to minimize the number of type changes for stability. While `bool` would work fine here, not sure if it makes sense to mix the use of the types in this module. Looking at the source, I see only one place where we've used `bool`, probably by accident...